code/modules/reagents/reagent_containers/food/snacks/egg.dm

```diff
@@ -93,9 +94,22 @@
 	name = "green egg"
 	desc = "Looks like it came from some genetically engineered chicken"
 	icon_state = "egg-vox"
+	can_color = FALSE
 
 /obj/item/weapon/reagent_containers/food/snacks/egg/vox/hatch()
 	visible_message("[src] hatches with a quiet cracking sound.")
 	new /mob/living/carbon/monkey/vox(get_turf(src))
 	processing_objects.Remove(src)
 	qdel(src)
+
+/obj/item/weapon/reagent_containers/food/snacks/egg/cockatrice
+	name = "cockatrice egg"
+	desc = "On the first glance it looks like a rock. Despite the myths, it's perfectly safe to touch and even eat it."
+	icon_state = "egg-cockatrice"
+	can_color = FALSE
+
+/obj/item/weapon/reagent_containers/food/snacks/egg/cockatrice/hatch()
+	visible_message("\The [src] hatches with a quiet cracking sound.")
+	new /mob/living/simple_animal/hostile/retaliate/cockatrice/chick(get_turf(src))
+	processing_objects.Remove(src)
+	qdel(src)
```

Comment:
rather than paste hatch(), instead add var/hatch_path to the normal egg, and change the original proc/hatch() to include
`new hatch_path(get_turf(src))`

Then you can just add `hatch_path = /mob/living/simple_animal/hostile/retaliate/cockatrice/chick` to this type of egg, and do the same for Vox eggs. More elegant, decreases file size, more modular for future egg types.